File: ros-beginners/rospy_simple_service_client/ros/script/simple_client.py
#!/usr/bin/env python

"""
This script is a simple python node which calls ros service
rospy_word_count and send request to the server.
"""
#-*- encoding: utf-8 -*-
__author__ = 'mohammad wasil'

import rospy
from ros_custom_msgs.srv import CountWords, CountWordsRequest

if __name__ == '__main__':

  #you are not necessarilt inilitialize node
  #since service client does not have to be a node
  rospy.init_node("rospy_simple_client_node", anonymous=False)
  rate = rospy.Rate(10)

  while not rospy.is_shutdown():
    rospy.wait_for_service('rospy_word_count')
    try:
      word_count = rospy.ServiceProxy('rospy_word_count', CountWords)

      req = CountWordsRequest()
      req.words.data =  "a test from the client node for a simple server client"
      req.count_articles.data = True

      resp = word_count.call(req)

      rospy.loginfo("Result -> Num of words: %s"%resp.num_words.data)
      rospy.loginfo("Result -> Num of articles: %s"%resp.num_articles.data)

    except rospy.ServiceException, e:
      print "Service call failed: %s"%e

    rate.sleep()

File: ros-beginners/roscpp_simple_service_client/ros/src/simple_client.cpp
#include <ros/ros.h>
#include <ros_custom_msgs/CountWords.h>

int main (int argc, char **argv)
{
  ros::init(argc, argv, "roscpp_simple_client_node");
  ros::NodeHandle nh;
  ros::ServiceClient service_client = nh.serviceClient<ros_custom_msgs::CountWords>("roscpp_word_count");
  ros::Rate loop_rate(10);
  while(ros::ok())
  {
    ros_custom_msgs::CountWords srv;
    srv.request.words.data = "A sample request from a client for a simple service client test";
    srv.request.count_articles.data = true;

    if (service_client.call(srv))
    {
      ROS_INFO_STREAM("Result -> Num of words: "<<srv.response.num_words.data);
      ROS_INFO_STREAM("Result -> Num of articles: "<<srv.response.num_articles.data);
    }
    else
    {
      ROS_ERROR("Failed to call the service server");
    }

    loop_rate.sleep();
  }
  return 0;
}
